src/quote/flashmint/hyeth/provider.test.ts

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { AddressZero, EthAddress } from 'constants/addresses'
+import { noopSwapData } from 'constants/swapdata'
+import { USDC, WETH } from 'constants/tokens'
+import { wei } from 'utils/numbers'
+import { Exchange } from 'utils'
+
+import {
+  IndexZeroExSwapQuoteProvider,
+  LocalhostProviderUrl,
+  QuoteTokens,
+} from 'tests/utils'
+
+import { FlashMintHyEthQuoteProvider } from './provider'
+
+const rpcUrl = LocalhostProviderUrl
+const swapQuoteProvider = IndexZeroExSwapQuoteProvider
+
+const { eth, hyeth, usdc, weth } = QuoteTokens
+const indexToken = hyeth
+
+describe('FlashMintHyEthQuoteProvider()', () => {
+  test('returns a quote for minting w/ ETH', async () => {
+    const request = {
+      isMinting: true,
+      inputToken: eth,
+      outputToken: {
+        symbol: indexToken.symbol,
+        decimals: 18,
+        address: indexToken.address!,
+      },
+      indexTokenAmount: wei(1).toBigInt(),
+      slippage: 0.5,
+    }
+    const quoteProvider = new FlashMintHyEthQuoteProvider(
+      rpcUrl,
+      swapQuoteProvider
+    )
+    const quote = await quoteProvider.getQuote(request)
+    if (!quote) fail()
+    expect(quote.indexTokenAmount).toEqual(request.indexTokenAmount)
+    expect(quote.inputOutputTokenAmount > 0).toBe(true)
+    const componentSwapDataIssue = [
+      noopSwapData,
+      noopSwapData,
+      noopSwapData,
+      noopSwapData,
+      noopSwapData,
+    ]
+    expect(quote.componentsSwapData).toStrictEqual(componentSwapDataIssue)
+    expect(quote.swapDataEthToInputOutputToken).toBeNull()
+    expect(quote.swapDataInputTokenToEth).toBeNull()
+  })
+
+  test('returns a quote for minting w/ WETH', async () => {
+    const request = {
+      isMinting: true,
+      inputToken: weth,
+      outputToken: {
+        symbol: indexToken.symbol,
+        decimals: 18,
+        address: indexToken.address!,
+      },
+      indexTokenAmount: wei(1).toBigInt(),
+      slippage: 0.5,
+    }
+    const quoteProvider = new FlashMintHyEthQuoteProvider(
+      rpcUrl,
+      swapQuoteProvider
+    )
+    const quote = await quoteProvider.getQuote(request)
+    if (!quote) fail()
+    expect(quote.indexTokenAmount).toEqual(request.indexTokenAmount)
+    expect(quote.inputOutputTokenAmount > 0).toBe(true)
+    const componentSwapDataIssue = [
+      noopSwapData,
+      noopSwapData,
+      noopSwapData,
+      noopSwapData,
+      noopSwapData,
```

Comment:
@pblivin0x @ckoopmann for now this will always just be those empty swap data tuples, right?

Reply:
yes 👍